opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -796,7 +796,11 @@ def start_span(  # pylint: disable=too-many-locals
         )
 
         if sampling_decision.sampled:
-            options = context.trace_flags | trace_api.TraceFlags.SAMPLED
+            # FIXME investigate this to make sure that this is the right
+            # approach. There seems to be an issue here currently because if
+            # context.trace_flags is 0, then options will be always set to 1.
+            # options = context.trace_flags | trace_api.TraceFlags.SAMPLED
+            options = context.trace_flags
```

Comment:
This is the same/similar issue as https://github.com/open-telemetry/opentelemetry-specification/issues/728 - it is intentional that the trace flags here should be consistent with the sampling decision - if the sampler says "Should sample" but the parent's trace flag's say "Do not sample", across all languages that I've looked at (java, go, python) we intentionally go with the sampler's decision preferentially.

Reply:
Yes, I was thinking the exact same thing 30 seconds ago. I'll revert this change.

Reply:
Hmm, I think @srjames90 raises a valid point. If we don't use the flag to decide if we sample or not we may end up with incompletely sampled traces when the spans pass a service boundary. Some services may decide to sample, others not and the active sampled flag will fly through them being ignored producing incompletely sampled traces, right? :man_shrugging: 

What do you think about this potential issue, @MitchellDumovic?

Reply:
I generally agree that this could be a problem, which is why I opened that issue to revisit the default sampler, which would fix that problem. I don't think that this is something we should decide here though.

Reply:
Ok, reverted it back to how it was before.

Reply:
Fixed